docs: :memo: add CONTRIBUTING from template (#62)
onlimit-study/dmp: CONTRIBUTING.md

# Contributing

## Issues and bugs :bug:

The easiest way to contribute is to report issues or bugs that you might
find while reading through the website. You can do this by creating a
[new](https://github.com/onlimit-study/dmp/issues/new/choose)
issue on our GitHub repository.

## Adding or modifying content :pencil2:

To contribute to `dmp`, you first need to install
[uv](https://docs.astral.sh/uv/) and
[justfile](https://just.systems/man/en/packages.html). We use uv and
justfile to manage our project, such as to run checks and test the
template. Both the uv and justfile websites have a more detailed guide
on using uv, but below are some simple instructions to get you started.

It's easiest to install uv and justfile using
[pipx](https://pypa.github.io/pipx/), so install that first. Then,
install uv and justfile by running:

``` bash
pipx install uv rust-just
```

We keep all our development workflows in the `justfile`, so you can
explore it to see what commands are available. To see a list of commands
available, run:

``` bash
just
```

> As you contribute, make sure your changes will pass our tests by opening

Reviewer: ```suggestion
As you contribute, make sure your changes will pass our checks by opening
```
Maybe?

a terminal so that the working directory is the root of this project's
repository and running:

``` bash
just run-all
```

> When committing changes, please try to follow [Conventional
> Commits](https://decisions.seedcase-project.org/why-conventional-commits)
> as Git messages. Using this convention allows us to be able to

Reviewer: ```suggestion
When committing changes, please follow the [Conventional
Commits specification](https://decisions.seedcase-project.org/why-conventional-commits)
in your commit messages. Using this convention allows us to be able to
```

> automatically create a release based on the commit message by using

Reviewer: ```suggestion
automatically create releases based on the commit message by using
```
